providers/src/airflow/providers/cncf/kubernetes/utils/pod_manager.py

```diff
@@ -565,6 +585,31 @@ def _reconcile_requested_log_containers(
             self.log.error("Could not retrieve containers for the pod: %s", pod_name)
         return containers_to_log
 
+    def fetch_requested_init_container_logs(
```

Comment:
does it really make sense to add a new method `fetch_requested_init_container_logs`?

is it not better to update `fetch_requested_container_logs` to support getting init containers?

Reply:
> does it really make sense to add a new method fetch_requested_init_container_logs?

Init containers runs sequencially and require different logic for fetching. For example it is better to fetch logs in order that they are definied

```
# sort by spec.initContainers because containers runs sequentially
containers_to_log = sorted(containers_to_log, key=lambda cn: all_containers.index(cn))
```

Reply:
Roger.  And to be clear my question is genuine and not a subtle way of saying "you must do this".

But, back to the matter, it looks like `fetch_requested_container_logs` does process the logs sequentially, doesn't it?  Looks like the only difference is you add "await start". Is that actually necessary? Why isn't it necessary with non-init containers?  Should we be doing it with non-init containers?

Reply:
For init containers, we only need to wait until the init container starts. After that, we can begin fetching logs. There may be cases where the first container successfully starts and returns logs, while the next one has not started yet. In such cases, we can start fetching logs as soon as possible.

I’m not sure why we have await_pod_start, but for init containers, this logic isn't necessary. That’s why we have a separate flow.

Reply:
I believe that fetching logs for non-init containers should be done in parallel, as we might have long-running containers with interactions where parallel logs could be useful. 

However, I suggest extracting this idea into a separate PR. :)

Reply:
Is that one ok @dstandish ?